horovod/common/response_cache.cc

```diff
@@ -139,8 +140,7 @@ void ResponseCache::put_(const Response& response, TensorParams& params) {
   bits_outdated_ = true;
 }
 
-void ResponseCache::put(const Response& response,
-                        const TensorTable& tensor_table) {
+void ResponseCache::put(const Response& response, TensorQueue& tensor_queue) {
```

Comment:
More we can make const, the better. We can make `const TensorQueue& tensor_queue` if we mark `GetTensorEntry` function as const.  Apply to as many things as you can.

Reply:
Guarding on a mutex will break the const qualifier. The most we can do is making `GetTensorEntry ` as const.